### ISSUE TITLE
Rule "no-unused-variable" fails to trigger in unused recursive functions

### ISSUE THREAD

**Issue by lcrespom** [palantir/tslint#1937]
### Bug Report
Rule `no-unused-variable` fails to trigger in unused recursive functions.

- __TSLint version__: `4.1.1`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: VS Code + TSLint extension v0.7.1.

#### TypeScript code being linted

```ts
// Example: typical recursive function
function factorial(x) {
    if ( x > 1)
        return x * factorial(x - 1);
    return 1;
}
```

with `tslint.json` configuration:

```json
    "no-unused-variable": true
```

#### Actual behavior
No lint error / warning is generated even if `factorial` is never used.

#### Expected behavior
An error should be generated if `factorial` is never used, but because it is used by itself, the linter incorrectly assumes it is already being used.

**Comment by andy-hanson** [comment]
Since this uses the TypeScript implementatino now (#2235), you should file a bug with TypeScript.
